### PR TITLE
Fix conditions that always evaluate to false

### PR DESCRIPTION
**Pull request by gwenneg**

### CODE REVIEW

File: devtools/maven/src/main/java/io/quarkus/maven/AddExtensionMojo.java
Comment (gwenneg) [inline devtools/maven/src/main/java/io/quarkus/maven/AddExtensionMojo.java]: @aloubyansky I need you to take a look at this one. Right now, removing this code doesn't change the plugin behaviour, but I'm pretty sure that's not the way we'd want to fix this class.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java
Comment (gwenneg) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java]: See line 44: `String uuid = BASE_ID + ERROR_COUNT.incrementAndGet();`

File: core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java
Comment (gwenneg) [inline core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java]: `getProxyInstance(returnType);` never returns `null`.

File: devtools/maven/src/main/java/io/quarkus/maven/AddExtensionMojo.java
Comment (aloubyansky) [reply devtools/maven/src/main/java/io/quarkus/maven/AddExtensionMojo.java]: Yes, it's not about just fixing this class, it's about fixing the tools. I think it's ok to merge it. The `buildFile` is closed in AddExtensions.

File: devtools/maven/src/main/java/io/quarkus/maven/AddExtensionMojo.java
Comment (gastaldi) [inline devtools/maven/src/main/java/io/quarkus/maven/AddExtensionMojo.java]:
```suggestion
                throw new MojoExecutionException("Failed to initialize the project build descriptor", e);
```

Comment (gwenneg) [reply devtools/maven/src/main/java/io/quarkus/maven/AddExtensionMojo.java]: Changed, thanks for the suggestion!

File: integration-tests/vault-app/src/main/java/io/quarkus/it/vault/VaultTestService.java
Comment (gsmet) [inline integration-tests/vault-app/src/main/java/io/quarkus/it/vault/VaultTestService.java]: This one makes sense but @vsevel could you have a look?

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java]: OK for these two.

File: core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java
Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java]: I'm not sold on this one.

**Review by gsmet — COMMENTED** [review]
I added a few comments.

File: core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java
Comment (gwenneg) [reply core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java]: Unless I'm missing something, the only value that can be returned from `getProxyInstance(returnType);` comes from https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java#L312-L313

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java]: It's not about missing something. It's just that doesn't cost anything to have it there so I would keep it.

File: integration-tests/vault-app/src/main/java/io/quarkus/it/vault/VaultTestService.java
Comment (vsevel) [reply integration-tests/vault-app/src/main/java/io/quarkus/it/vault/VaultTestService.java]: the second case should be between `password` and `expectedPassword` (not between `password` and `someSecret`)
`password` and `someSecret` are calculated, `expectedPassword` is the reference (could be a constant)
so something like this:
```
        String password = ConfigProviderResolver.instance().getConfig().getValue("password", String.class);
        if (!expectedPassword.equals(password)) {
            return "password=" + password + "; expected: " + expectedPassword;
        }
```
thanks for catching that one.

File: core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java
Comment (gwenneg) [reply core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java]: It's back with curly braces in bonus!

File: integration-tests/vault-app/src/main/java/io/quarkus/it/vault/VaultTestService.java
Comment (gwenneg) [reply integration-tests/vault-app/src/main/java/io/quarkus/it/vault/VaultTestService.java]: Thanks @vsevel! I changed the code according to your suggestion.

**Comment by gwenneg** [comment]
@gsmet It should be all good now.

**Comment by gwenneg** [comment]
@gastaldi Could you please do the approving review for this PR? I was waiting for @gsmet approval, but he's away for a few days and all changes have already been approved individually, so it shouldn't be a problem.

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by gwenneg** [comment]
Thanks!

**Pull request merged**
